src/java/org/apache/cassandra/config/CassandraRelevantProperties.java

```diff
@@ -500,6 +492,14 @@ public enum CassandraRelevantProperties
     SSTABLE_FORMAT_DEFAULT("cassandra.sstable.format.default"),
     START_GOSSIP("cassandra.start_gossip", "true"),
     START_NATIVE_TRANSPORT("cassandra.start_native_transport"),
+    /**
+     * {@link StorageCompatibilityMode} mode sets how the node will behave, sstable or messaging versions to use etc according to a yaml setting.
+     * But many tests don't load the config hence we need to force it otherwise they would run always under the default. Config is null for junits
+     * that don't load the config. Get from env var that CI/build.xml sets.
+     *
+     * This is a dev/CI only property. Do not use otherwise.
+     */
+    STORAGE_COMPATIBILITY_MODE("cassandra.test.storage_compatibility_mode", StorageCompatibilityMode.CASSANDRA_4.toString()),
```

Comment:
Imo this name is misleading now. I can see sbdy easily reading this variable expecting to get the SCM when they should in fact be getting it from config. I would rename it to sthg that discouraged it's usage or made it clear it's a test only hacky override:
- JUNIT_STORAGE_COMPATIBILITY_MODE
- STORAGE_COMPATIBILITY_MODE_FOR_TESTS
- JUNIT_STORAGE_COMPATIBILITY_MODE_TEST_OVERRIDE
- etc

wdyt?

Reply:
SGTM